src/load.ts

```diff
@@ -0,0 +1,75 @@
+import has from './has';
+import Promise from './Promise';
+
+declare var require: Function;
+declare var define: any;
+
+export interface Load {
+	(module: ModuleObject, ...moduleIds: string[]): Promise<any[]>;
+}
+
+export interface ModuleObject {
+	id: string;
+}
+
+function getBaseId(moduleObject: ModuleObject): string {
+	var id: string = moduleObject.id;
+	var index: number = id.lastIndexOf('/');
+
+	if (index > -1) {
+		id = id.substring(0, index);
+	}
+	return id;
+}
+
+interface HostLoad {
+	(moduleIds: string[]): Promise<any[]>;
+}
+
+const hostLoad: HostLoad = (function () {
+	if (typeof define === 'function' && define.amd) {
+		return function (moduleIds: string[]): Promise<any[]> {
+			return new Promise(function (resolve) {
+				// TODO: Error path
+				require(moduleIds, function (...modules: any[]) {
+					resolve(modules);
+				});
+			});
+		};
+	}
+	else if (has('host-node')) {
+		return function (moduleIds: string[]): Promise<any[]> {
+			return new Promise(function (resolve, reject) {
+				try {
+					resolve(moduleIds.map(function (moduleId): any {
+						return require(moduleId);
+					}));
+				}
+				catch (error) {
+					reject(error);
+				}
+			});
+		};
+	}
+})();
+
+const load: Load = (function () {
+	if (!hostLoad) {
+		return function () {
+			return Promise.reject(new Error('Unknown loader'));
+		};
+	}
+	else {
+		return function (module: ModuleObject, ...moduleIds: string[]): Promise<any> {
```

Comment:
I chose `module` instead of `require` because, according to the AMD spec, there's no way to get the absolute module ID given a relative module ID. This means that while passing `require` from Node and both the Dojo 1 and 2 loaders would work fine, `require` from RequireJS would be pretty much useless.